qa/rpc-tests/mempool_size_limit.py

```diff
@@ -0,0 +1,593 @@
+        outputs[addr] = self.satoshi_round(send_value)
+        rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+        signresult = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE")
+        csize = len(signresult['hex']) // 2
+
+        txFile = open(self.options.tmpdir + "/ctxs/ctx0", "wb")
+        txFile.write(bytes(str((low_fee) / csize) + '\n', encoding='utf8'))
+        txFile.write(bytes(signresult["hex"], encoding='utf8'))
+        txFile.close()
+        txs.append({'fee': high_fee / csize, 'hex': signresult["hex"]})
+
+        ptx = self.nodes[0].decoderawtransaction(signresult["hex"])
+
+        assert(high_fee < t['amount'])
+        inputs = []
+        inputs.append({ "txid" : ptx['txid'], "vout" : 0})
+        outputs = {}
+        send_value = t['amount'] - low_fee - high_fee
+        outputs[self.nodes[0].getnewaddress()] = self.satoshi_round(send_value)
+        rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+        signresult = self.nodes[0].signrawtransaction(rawtx, [{"txid": ptx['txid'], "vout": 0, "scriptPubKey": ptx['vout'][0]['scriptPubKey']['hex']}], None, "NONE")
+        csize = len(signresult['hex']) // 2
+
+        txFile = open(self.options.tmpdir + "/ctxs/ctx1", "wb")
+        txFile.write(bytes(str(high_fee / csize) + '\n', encoding='utf8'))
+        txFile.write(bytes(signresult["hex"], encoding='utf8'))
+        txFile.close()
+        txs.append({'fee': high_fee / csize, 'hex': signresult["hex"]})
+
+        return txs
+
+    def create_big_transactions(self, size):
+        self.create_tx_script()
+        addr = self.nodes[0].getnewaddress()
+        txs = []
+        tot_size = 0
+        i = 0
+        os.makedirs(self.options.tmpdir + "/txs")
+        while (tot_size < size):
+            t = self.utxos.pop()
+            inputs = []
+            inputs.append({ "txid" : t["txid"], "vout" : t["vout"]})
+            outputs = {}
+            send_value = t['amount']
+            outputs[addr] = self.satoshi_round(send_value)
+            rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+            out_num_pos = rawtx.find('ffffffff')
+            script_position = rawtx.find('76a914')
+            script_length = int(rawtx[script_position - 2 : script_position], 16) * 2
+
+            newtx = rawtx[0:out_num_pos + 8]
+            newtx = newtx + self.txouts
+            newtx = newtx + rawtx[script_position + script_length:]
+
+            signresult = self.nodes[0].signrawtransaction(newtx, None, None, "NONE")
+            tsize = len(signresult['hex']) // 2
+            tot_size += tsize
+            txFile = open(self.options.tmpdir + "/txs/tx" + str(i), "wb")
+            txFile.write(bytes(str(t['amount'] / tsize) + '\n', encoding='utf8'))
+            txFile.write(bytes(signresult["hex"], encoding='utf8'))
+            txFile.close()
+            txs.append({'fee': t['amount'] / tsize, 'hex': signresult["hex"]})
```

Comment:
I found a way to exactly compute fee rate during data setup
```suggestion
txs.append({'fee': (t['amount'] - 128 * 5816) / tsize, 'hex': signresult["hex"]}) # 128 is outputs quantity, 5816 (xb816 reversed) is output value
```
Probably it would be good also to refactor the key name, moving from 'fee' to 'feerate', also to avoid confusion with the key 'fee' of `create_sc_certificates`, which actually refers to fee itself (not fee rate)
With above computation, I think we can also make simpler the `run_test` function, avoiding to distinguish between `tx_input` and `feerate` (and their corresponding minimum), keeping only fee rate.

Reply:
Agreed to handle this in a separate PR.